frame/system/src/lib.rs

```diff
@@ -552,8 +558,9 @@ decl_module! {
 		/// - `O(C)` where `C` length of `code`
 		/// - 1 storage write (codec `O(C)`).
 		/// - 1 event.
+		/// The weight of this function is dependent on the runtime. We will treat this as a full block.
 		/// # </weight>
-		#[weight = (200_000_000, DispatchClass::Operational)]
+		#[weight = (crate::Module::<T>::max_extrinsic_weight(DispatchClass::Operational), DispatchClass::Operational)]
```

Comment:
Should `set_code_without_checks` have `DispatchClass::Mandatory`?

Please triple check `max_extrinsic_weight` logic